content/en/blog/2024/spring-starter-stable/index.md

```diff
@@ -120,3 +120,505 @@ We came up with a new interface,
+
+First, we clone the Spring PetClinic application from GitHub:
+
+```bash
+git clone https://github.com/spring-projects/spring-petclinic.git
+```
+
+In the `pom.xml` file, let's add the OpenTelemetry instrumentation BOM:
+```xml
+  <dependencyManagement>
+    <dependencies>
+
+        <dependency>
+            <groupId>io.opentelemetry.instrumentation</groupId>
+            <artifactId>opentelemetry-instrumentation-bom</artifactId>
+            <version>2.7.0</version>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+```
+
+Now, we can include the OpenTelemetry Spring Boot starter dependency to the Spring PetClinic application:
+
+```xml
+  <dependency>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-spring-boot-starter</artifactId>
+  </dependency>
+```
+
+Navigate to the project directory and build a Spring Boot native image application:
+```bash
+cd spring-petclinic
+mvn -Pnative spring-boot:build-image -Dspring-boot.build-image.imageName=spring-petclinic-native
+```
+
+You may have to disable the `PostgresIntegrationTests` test class to get this command line, [see](https://github.com/spring-projects/spring-petclinic/issues/1522).
+
+The OpenTelemetry Spring Boot starter sends the telemetry data with the [OpenTelemetry Protocol](https://opentelemetry.io/docs/specs/otlp/) (OTLP). By default, it sends the data over HTTP. You can also switch to gRPC.
+
+By default, the OpenTelemetry Spring Boot starter sends telemetry data using the [OpenTelemetry Protocol](https://opentelemetry.io/docs/specs/otlp/) (OTLP) over HTTP, but it can also use gRPC.
+
+We are going to add an [OpenTelemetry collector](https://opentelemetry.io/docs/collector/) and display the telemetry data in the collector logs.
+
+To do this, let's add the following `docker-compose-otel.yml` and `collector-spring-native-config.yaml` files in the `spring-petclinic` directory: 
+
+docker-compose-otel.yml
+```yaml
+version: '3.8'
+services:
+  app:
+    image: spring-petclinic-native
+    environment:
+      OTEL_SERVICE_NAME: "graal-native-example-app"
+      OTEL_EXPORTER_OTLP_ENDPOINT: "http://collector:4318"
+    ports:
+      - "8080:8080"
+    depends_on:
+      - collector
+  collector:
+    image: otel/opentelemetry-collector-contrib:0.109.0
+    volumes:
+      - ./collector-spring-native-config.yaml:/collector-spring-native-config.yaml
+    command: ["--config=/collector-spring-native-config.yaml"]
+    expose:
+      - "4317"
+    ports:
+      - "4317:4317"
+```
+
+collector-spring-native-config.yaml
+```yaml 
+receivers:
+  otlp:
+    protocols:
+      http:
+        endpoint: "0.0.0.0:4318"
+exporters:
+  logging:
+    verbosity: detailed
+service:
+  pipelines:
+    metrics:
+      receivers: [otlp]
+      exporters: [logging]
```

Comment:
logging is outdated: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/debugexporter/README.md

Reply:
It does not seem applicable to the pipelines section.

Reply:
i think these are changing, see here https://github.com/open-telemetry/opentelemetry-java-examples/pull/491

so i think this becomes:
```
exporters:
  debug:
    verbosity: detailed
service:
  pipelines:
    metrics:
      receivers: [otlp]
      exporters: [debug]
      ...
```